server/models/k8s_context.go

```diff
@@ -329,3 +330,61 @@ func (kc *K8sContext) AssignServerID() error {
 
 	return nil
 }
+
+// FlushMeshSyncData will flush the meshsync data for the passed kubernetes contextID
+func FlushMeshSyncData(ctx context.Context, ctxID string, provider Provider) error {
+	// Gets all the available kubernetes contexts
+	k8sctxs, ok := ctx.Value(AllKubeClusterKey).([]K8sContext)
+	if !ok || len(k8sctxs) == 0 {
+		return ErrEmptyCurrentK8sContext
+	}
+	var sid string
+	// Gets the serverID for the passed contextID
+	for _, k8ctx := range k8sctxs {
+		if k8ctx.ID == ctxID && k8ctx.KubernetesServerID != nil {
+			sid = k8ctx.KubernetesServerID.String()
+			break
```

Comment:
Instead of looping again, add that extra check and increment the reference counter here only

Reply:
gotcha, making change